src/include/configuration/configuration_manager.h

```diff
@@ -0,0 +1,77 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// configuration_manager.h
+//
+// Identification: src/include/configuration/configuration_manager.h
+//
+// Copyright (c) 2015-2017, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+
+#pragma once
+
+#include <map>
+#include "type/types.h"
+#include "type/value.h"
+#include "common/exception.h"
+#include "common/printable.h"
+#include "configuration/configuration_type.h"
+
+namespace peloton {
+namespace configuration {
+
+class ConfigurationManager : public Printable {
+
+  friend class ConfigurationUtil;
+
+public:
+  static ConfigurationManager* GetInstance();
+
+  void InitializeCatalog();
+
+  const std::string GetInfo() const;
+
+  void ShowInfo();
+
+private:
+
+  struct Param {
+    std::string name;
+    void* gflags_ptr;
+    std::string desc;
+    type::Value value, default_value;
+    bool is_mutable, is_persistent;
+    Param() {}
```

Comment:
It's only needed if you use `operator[]`. Remove the default constructor and don't use the `operator[]` to keep the interface clean. Replace searches with `config.find()` and insertions with `config.insert()`, or even better `config.emplace()`.